lib/msal-node/src/config/ManagedIdentityId.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    DEFAULT_MANAGED_IDENTITY_ID,
+    ManagedIdentityIdType,
+} from "../utils/Constants";
+
+export class ManagedIdentityId {
```

Comment:
Since this is JS and ppl like JSON, I think you should have an object model that is more declarative and less static-y. 
```json
{ 
    "systemAssigned": true
}
```
```json
{ 
    "systemAssigned": false,
    "userAssignedClientId": 12345
}
```

```json
{ 
    "systemAssigned": false,
    "userAssignedObjectId": 12345
}
```

Reply:
@rayluo for a perspective, as I think this is what we did in MSAL PY as well.

Reply:
I've started implementing this. I think this is more error prone as the developer could input all three of userAssignedClientId, userAssignedResourceId and userAssignedObjectId.I can if statements to mitigate this, but it will be much more code than before.

Reply:
Ok, how about this then. Let's support the JSON approach for the most common use case. I believe 99% of app developers rely on system assigned identity or on user assigned identity + client_id. 

```json
{ 
    "systemAssigned": true
}
```

```json
{ 
    "systemAssigned": false, // optional, but if set, it can't be true
    "userAssignedIdenityClientId": 12345
}
```

The 1% needing user assigned identity by resourceId can use the static method.

Reply:
MSAL .NET is not JSON driven, but MSAL Node is and Id.Web are. So proposal is more like: https://github.com/AzureAD/microsoft-identity-web/issues/1775#issuecomment-1769864869

Agree on hiding "IdType" - this is implementation detail. 

@Robbie-Microsoft - can you please organize a meeting and include all reviewers and @localden and discuss the public API you have ? In particular we need to see how to specify SAMI and UAMI.

Reply:
I've implemented this